pow/src/main/java/tech/pegasys/teku/pow/fallback/strategy/PriorityEth1ProviderSelector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow.fallback.strategy;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import tech.pegasys.teku.pow.Eth1Provider;
+
+public class PriorityEth1ProviderSelector implements Eth1ProviderSelector {
+
+  final List<Eth1Provider> candidates;
+  private final AtomicReference<Eth1Provider> bestCandidate;
+  private final AtomicInteger bestCandidateIndex;
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public PriorityEth1ProviderSelector(final List<Eth1Provider> candidates) {
+    assert candidates != null && !candidates.isEmpty();
+    this.candidates = candidates;
+    this.bestCandidateIndex = new AtomicInteger(0);
+    this.bestCandidate = new AtomicReference<>(this.candidates.get(this.bestCandidateIndex.get()));
+  }
+
+  @Override
+  public Eth1Provider bestCandidate() {
+    return bestCandidate.get();
+  }
+
+  @Override
+  public void updateBestCandidate() {
+    int nextIndex = (bestCandidateIndex.get() + 1) % candidates.size();
+    bestCandidateIndex.set(nextIndex);
+    bestCandidate.set(candidates.get(bestCandidateIndex.get()));
+  }
```

Comment:
This is still setup so that it will get "stuck" using a backup node.  For example if I'm using a local Eth1 node first and then Infura second, the first time my Eth1 node returns an error, the best candidate will be updated to use Infura then because Infura is reliable it is unlikely to get any further errors and will just always use Infura but we actually want it to switch back to using the local node as soon as its available again.

I would probably drop the concept of best candidate and just have this return a list of available providers (in follow up PRs it could filter them based on whether its in-sync or not, on the right chain or has returned errors recently etc).  That would also fix the race condition in `FallbackAwareEth1Provider` if it was just iterating through the list.

Reply:
I am now using an iterator to iterate over the list of available providers, restarting always with the first.

Reply:
I dropped the concept of best candidate